tests/zydis_encoder_types.py

```diff
@@ -0,0 +1,2171 @@
+    'ZYDIS_ENCODABLE_PREFIX_BRANCH_NOT_TAKEN',
+    'ZYDIS_ENCODABLE_PREFIX_BRANCH_TAKEN',
+    'ZYDIS_ENCODABLE_PREFIX_NOTRACK',
+    'ZYDIS_ENCODABLE_PREFIX_SEGMENT_CS',
+    'ZYDIS_ENCODABLE_PREFIX_SEGMENT_SS',
+    'ZYDIS_ENCODABLE_PREFIX_SEGMENT_DS',
+    'ZYDIS_ENCODABLE_PREFIX_SEGMENT_ES',
+    'ZYDIS_ENCODABLE_PREFIX_SEGMENT_FS',
+    'ZYDIS_ENCODABLE_PREFIX_SEGMENT_GS',
+], start=0)
+
+
+ZydisEncodableBranchType = IntEnum('ZydisEncodableBranchType', [
+    'ZYDIS_ENCODABLE_BRANCH_TYPE_NONE',
+    'ZYDIS_ENCODABLE_BRANCH_TYPE_SHORT',
+    'ZYDIS_ENCODABLE_BRANCH_TYPE_NEAR16',
+    'ZYDIS_ENCODABLE_BRANCH_TYPE_NEAR32',
+    'ZYDIS_ENCODABLE_BRANCH_TYPE_NEAR64',
+    'ZYDIS_ENCODABLE_BRANCH_TYPE_FAR16',
+    'ZYDIS_ENCODABLE_BRANCH_TYPE_FAR32',
+    'ZYDIS_ENCODABLE_BRANCH_TYPE_FAR64',
+], start=0)
+
+
+ZydisAddressSizeHint = IntEnum('ZydisAddressSizeHint', [
+    'ZYDIS_ADDRESS_SIZE_HINT_NONE',
+    'ZYDIS_ADDRESS_SIZE_HINT_16',
+    'ZYDIS_ADDRESS_SIZE_HINT_32',
+    'ZYDIS_ADDRESS_SIZE_HINT_64',
+], start=0)
+
+
+ZydisOperandSizeHint = IntEnum('ZydisOperandSizeHint', [
+    'ZYDIS_OPERAND_SIZE_HINT_NONE',
+    'ZYDIS_OPERAND_SIZE_HINT_8',
+    'ZYDIS_OPERAND_SIZE_HINT_16',
+    'ZYDIS_OPERAND_SIZE_HINT_32',
+    'ZYDIS_OPERAND_SIZE_HINT_64',
+], start=0)
+
+
+ZydisBroadcastMode = IntEnum('', [
+    'ZYDIS_BROADCAST_MODE_INVALID',
+    'ZYDIS_BROADCAST_MODE_1_TO_2',
+    'ZYDIS_BROADCAST_MODE_1_TO_4',
+    'ZYDIS_BROADCAST_MODE_1_TO_8',
+    'ZYDIS_BROADCAST_MODE_1_TO_16',
+    'ZYDIS_BROADCAST_MODE_1_TO_32',
+    'ZYDIS_BROADCAST_MODE_1_TO_64',
+    'ZYDIS_BROADCAST_MODE_2_TO_4',
+    'ZYDIS_BROADCAST_MODE_2_TO_8',
+    'ZYDIS_BROADCAST_MODE_2_TO_16',
+    'ZYDIS_BROADCAST_MODE_4_TO_8',
+    'ZYDIS_BROADCAST_MODE_4_TO_16',
+    'ZYDIS_BROADCAST_MODE_8_TO_16',
+], start=0)
+
+
+ZydisRoundingMode = IntEnum('ZydisRoundingMode', [
+    'ZYDIS_ROUNDING_MODE_INVALID',
+    'ZYDIS_ROUNDING_MODE_RN',
+    'ZYDIS_ROUNDING_MODE_RD',
+    'ZYDIS_ROUNDING_MODE_RU',
+    'ZYDIS_ROUNDING_MODE_RZ',
+], start=0)
+
+
+ZydisConversionMode = IntEnum('ZydisConversionMode', [
+    'ZYDIS_CONVERSION_MODE_INVALID',
+    'ZYDIS_CONVERSION_MODE_FLOAT16',
+    'ZYDIS_CONVERSION_MODE_SINT8',
+    'ZYDIS_CONVERSION_MODE_UINT8',
+    'ZYDIS_CONVERSION_MODE_SINT16',
+    'ZYDIS_CONVERSION_MODE_UINT16',
+], start=0)
+
+
+ZydisSwizzleMode = IntEnum('ZydisSwizzleMode', [
+    'ZYDIS_SWIZZLE_MODE_INVALID',
+    'ZYDIS_SWIZZLE_MODE_DCBA',
+    'ZYDIS_SWIZZLE_MODE_CDAB',
+    'ZYDIS_SWIZZLE_MODE_BADC',
+    'ZYDIS_SWIZZLE_MODE_DACB',
+    'ZYDIS_SWIZZLE_MODE_AAAA',
+    'ZYDIS_SWIZZLE_MODE_BBBB',
+    'ZYDIS_SWIZZLE_MODE_CCCC',
+    'ZYDIS_SWIZZLE_MODE_DDDD',
+], start=0)
+
+
+ZydisOperandType = IntEnum('ZydisOperandType', [
+    'ZYDIS_OPERAND_TYPE_UNUSED',
+    'ZYDIS_OPERAND_TYPE_REGISTER',
+    'ZYDIS_OPERAND_TYPE_MEMORY',
+    'ZYDIS_OPERAND_TYPE_POINTER',
+    'ZYDIS_OPERAND_TYPE_IMMEDIATE',
+], start=0)
+
+
+ZydisMnemonic = IntEnum('ZydisMnemonic', [
```

Comment:
Possible future improvement: at least make `ZydisMnemonic` and `ZydisRegister` auto-generated. Ideally whole file can be auto-generated. There are two ways to go about it:
- use existing generator to do heavy-lifting here (good for mnemonics and registers, but not for other enums)
- at the cost of extra python dependency it's possible to parse C header files and generate all types at run-time

Reply:
Very nice work on the regression tests :-) 

I personally prefer the second option here as it reduces the amount of places where we have to update enums (especially the ones different to `ZydisMnemonic` and `ZydisRegister`). The current generator would only be able to auto-generate `ZydisMnemonic`. For `ZydisRegister` there is a second generator, but I did never finish it (it was supposed to generate the `C` enums and as well the `Delphi` ones for the primary generator from a `json` file).